qemu/tests/nvdimm.py

```diff
@@ -0,0 +1,141 @@
+import logging
+
+from autotest.client.shared import error
+from autotest.client import utils
+
+
+class NvdimmTest(object):
+    """
+    Class for NVDIMM test
+    """
+    def __init__(self, params, env):
+        """
+        Init the default values of NvdimmTest object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.session = {}
+        self.params = params
+        self.env = env
+
+    def check_nvdimm(self):
+        """
+        Check nvdimm in guest
+        """
+        check_cmd = "test -b %s" % self.params.get("pmem", "/dev/pmem0")
+        status, output = self.session.cmd_status_output(check_cmd, timeout=10)
+        if status != 0:
+            raise error.TestFail("check nvdimm in guest failed: %s" % output)
+
+    def verify_nvdimm(self, vm):
+        """
+        verify nvdimm in monitor
+
+        :params vm: VM object
+        """
+        dev_ids = map(lambda x: "-".join(["dimm", x]),
+                      self.params.objects('mem_devs'))
+        for info in vm.monitor.info("memory-devices"):
+            if str(info['data']['id']) not in dev_ids:
+                return False
+        return True
```

Comment:
"return True" should be in the for loop?

Maybe:
for ****:
    if *** in dev_ids:
        return True
return False